### PR TITLE
feat: upgrade apis to use flink 1.14

### PR DESCRIPTION
**Pull request by arujit**
- Upgrading to the most recent stable flink version (1.14).
- Fixing deprecated APIs and do the required refactoring.

### CODE REVIEW

File: dagger-common/build.gradle [hunk 1]
```diff
@@ -57,7 +57,9 @@ dependencies {
     compileOnly group: 'org.apache.flink', name: 'flink-table-common', version: flinkVersion
     compileOnly group: 'org.apache.flink', name: 'flink-table', version: flinkVersion
     compileOnly group: 'org.apache.flink', name: 'flink-table-api-java-bridge_2.11', version: flinkVersion
-    
+    compileOnly group: 'org.apache.flink', name: 'flink-connector-kafka_2.11', version: flinkVersion
+
+    dependenciesCommonJar 'org.apache.flink:flink-metrics-dropwizard:' + flinkVersion
```
Comment (prakharmathur82) [inline dagger-common/build.gradle:62]: it is duplicated

File: dagger-common/src/main/java/io/odpf/dagger/common/configuration/Configuration.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package io.odpf.dagger.common.configuration;
+
+import org.apache.flink.api.java.utils.ParameterTool;
+
+import java.io.Serializable;
+
+public class Configuration implements Serializable {
```
Comment (prakharmathur82) [inline dagger-common/src/main/java/io/odpf/dagger/common/configuration/Configuration.java:7]: we should add tests for this class, as there is custom logic in getters

File: dagger-common/src/test/java/io/odpf/dagger/common/watermark/StreamWatermarkAssignerTest.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    public void shouldAssignTimestampAndWatermarksToKafkaConsumer() {
+        LastColumnWatermark lastColumnWatermark = new LastColumnWatermark();
+        StreamWatermarkAssigner streamWatermarkAssigner = new StreamWatermarkAssigner(lastColumnWatermark);
+        streamWatermarkAssigner.consumerAssignTimeStampAndWatermark(consumer, 10L, true);
+
+        verify(consumer, times(1)).assignTimestampsAndWatermarks(any(WatermarkStrategy.class));
+    }
+
+    @Test
+    public void shouldNotAssignTimestampAndWatermarksToKafkaConsumerIfPerPartitionWatermarkDisabled() {
```
Comment (prakharmathur82) [inline dagger-common/src/test/java/io/odpf/dagger/common/watermark/StreamWatermarkAssignerTest.java:51]: Redundant test

File: dagger-common/src/test/java/io/odpf/dagger/common/watermark/StreamWatermarkAssignerTest.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        LastColumnWatermark lastColumnWatermark = new LastColumnWatermark();
+        StreamWatermarkAssigner streamWatermarkAssigner = new StreamWatermarkAssigner(lastColumnWatermark);
+        streamWatermarkAssigner.consumerAssignTimeStampAndWatermark(consumer, 10L, true);
+
+        verify(consumer, times(1)).assignTimestampsAndWatermarks(any(WatermarkStrategy.class));
+    }
+
+    @Test
+    public void shouldAssignTimestampAndWatermarksToSource() {
```
Comment (prakharmathur82) [inline dagger-common/src/test/java/io/odpf/dagger/common/watermark/StreamWatermarkAssignerTest.java:60]: test name is wrong

File: dagger-functions/src/test/java/io/odpf/dagger/functions/transformers/FeatureTransformerTest.java [hunk 5]
```diff
@@ -24,6 +25,9 @@
     @Mock
     private Configuration configuration;
 
+    public FeatureTransformerTest() {
```
Comment (prakharmathur82) [inline dagger-functions/src/test/java/io/odpf/dagger/functions/transformers/FeatureTransformerTest.java:28]: what is this for?

Comment (arujit) [reply dagger-functions/src/test/java/io/odpf/dagger/functions/transformers/FeatureTransformerTest.java:28]: Missed this. Not being used anywhere. Removing this.

**Pull request merged**